### PR TITLE
Dependency: remove golang/protobuf, update rules_go

### PR DESCRIPTION
**Pull request by moderation**
Description: Removes golang/protobuf dependency. This is an old @sesmith177 TODO and now the version of `rules_go` we import has the requisite well known types (WKT). Bumping `rules_go` to the latest version 0.18.4 ([release notes](https://github.com/bazelbuild/rules_go/releases/tag/0.18.4)) aligns with PGV which is on 0.18.3.
Risk Level: low (I think but there are WKT changes)
Testing: `bazel test //test/...`
Docs Changes: none required
Release Notes: none required

### CODE REVIEW

**Comment by lizan** [comment]
the `api` CI failure is real, seems like go code doesn't compile?

**Comment by moderation** [comment]
@lizan it is real and I'm reproducing locally. Can't quite figure out where it is coming from my. Might be PGV. Will keep digging today. Hoping maybe @htuch which his Bazel foo will know what is happening.
```
/home/moderation/.cache/bazel/_bazel_moderation/51ef416c18963c97653653fae6a84309/sandbox/linux-sandbox/2/execroot/envoy/bazel-out/k8-fastbuild/bin/external/envoy_api/envoy/type/matcher/linux_amd64_stripped/value_go_proto%/github.com/envoyproxy/data-plane-api/api/value/value.pb.go:39:33: use of package strin
g without selector
/home/moderation/.cache/bazel/_bazel_moderation/51ef416c18963c97653653fae6a84309/sandbox/linux-sandbox/2/execroot/envoy/bazel-out/k8-fastbuild/bin/external/envoy_api/envoy/type/matcher/linux_amd64_stripped/value_go_proto%/github.com/envoyproxy/data-plane-api/api/value/value.pb.go:164:10: impossible type ass
ertion:
        *ValueMatcher does not implement proto.Message (wrong type for String method)
                have String() <T>
                want String() string
/home/moderation/.cache/bazel/_bazel_moderation/51ef416c18963c97653653fae6a84309/sandbox/linux-sandbox/2/execroot/envoy/bazel-out/k8-fastbuild/bin/external/envoy_api/envoy/type/matcher/linux_amd64_stripped/value_go_proto%/github.com/envoyproxy/data-plane-api/api/value/value.pb.go:209:10: impossible type ass
ertion:
        *ValueMatcher does not implement proto.Message (wrong type for String method)
                have String() <T>
                want String() string
/home/moderation/.cache/bazel/_bazel_moderation/51ef416c18963c97653653fae6a84309/sandbox/linux-sandbox/2/execroot/envoy/bazel-out/k8-fastbuild/bin/external/envoy_api/envoy/type/matcher/linux_amd64_stripped/value_go_proto%/github.com/envoyproxy/data-plane-api/api/value/value.pb.go:263:10: impossible type ass
ertion:
        *ValueMatcher does not implement proto.Message (wrong type for String method)
                have String() <T>
                want String() string
/home/moderation/.cache/bazel/_bazel_moderation/51ef416c18963c97653653fae6a84309/sandbox/linux-sandbox/2/execroot/envoy/bazel-out/k8-fastbuild/bin/external/envoy_api/envoy/type/matcher/linux_amd64_stripped/value_go_proto%/github.com/envoyproxy/data-plane-api/api/value/value.pb.go:306:43: use of package stri
ng without selector
/home/moderation/.cache/bazel/_bazel_moderation/51ef416c18963c97653653fae6a84309/sandbox/linux-sandbox/2/execroot/envoy/bazel-out/k8-fastbuild/bin/external/envoy_api/envoy/type/matcher/linux_amd64_stripped/value_go_proto%/github.com/envoyproxy/data-plane-api/api/value/value.pb.go:339:32: use of package stri
ng without selector
/home/moderation/.cache/bazel/_bazel_moderation/51ef416c18963c97653653fae6a84309/sandbox/linux-sandbox/2/execroot/envoy/bazel-out/k8-fastbuild/bin/external/envoy_api/envoy/type/matcher/linux_amd64_stripped/value_go_proto%/github.com/envoyproxy/data-plane-api/api/value/value.pb.go:394:10: impossible type ass
ertion:
        *ListMatcher does not implement proto.Message (wrong type for String method)
                have String() <T>
                want String() string
/home/moderation/.cache/bazel/_bazel_moderation/51ef416c18963c97653653fae6a84309/sandbox/linux-sandbox/2/execroot/envoy/bazel-out/k8-fastbuild/bin/external/envoy_api/envoy/type/matcher/linux_amd64_stripped/value_go_proto%/github.com/envoyproxy/data-plane-api/api/value/value.pb.go:410:10: impossible type as$
ertion:
        *ListMatcher does not implement proto.Message (wrong type for String method)
                have String() <T>
                want String() string
/home/moderation/.cache/bazel/_bazel_moderation/51ef416c18963c97653653fae6a84309/sandbox/linux-sandbox/2/execroot/envoy/bazel-out/k8-fastbuild/bin/external/envoy_api/envoy/type/matcher/linux_amd64_stripped/value_go_proto%/github.com/envoyproxy/data-plane-api/api/value/value.pb.go:426:10: impossible type as$
ertion:
        *ListMatcher does not implement proto.Message (wrong type for String method)
                have String() <T>
                want String() string
```

**Comment by htuch** [comment]
@moderation not really sure, but you might want to make sure PGV (and protoc-gen-star) is also consistently consuming these types.

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -137,7 +137,10 @@ REPOSITORY_LOCATIONS = dict(
         strip_prefix = "libevent-1cd8830de27c30c5324c75bfb6012c969c09ca2c",
         urls = ["https://github.com/libevent/libevent/archive/1cd8830de27c30c5324c75bfb6012c969c09ca2c.tar.gz"],
     ),
-    com_github_madler_zlib = dict(
+    net_zlib = dict(
+        # TODO(moderation): revert to com_github_madler_zlib name pending resolution of workaround
```
Comment (lizan) [inline bazel/repository_locations.bzl]: Not sure we need the revert, if `net_zlib` is commonly used in protobuf and/or rules_go, we should just follow it. zlib.net is the canonical URL of the lib per its doc.

Comment (moderation) [reply bazel/repository_locations.bzl]: Make sense. Are we OK with leaving the download target as Github? Seems like the safer / reliable option as I'm not sure where zlib.net is hosted.

**Pull request merged**